CSharp/BatchExplorer/Service/BatchService.cs

```diff
@@ -201,13 +201,48 @@ public Task<CloudPool> GetPoolAsync(string poolId)
             int targetDedicated,
             TimeSpan? timeout,
             ComputeNodeDeallocationOption? deallocationOption)
-        {
+        {
+            var pool = await this.GetPoolAsync(poolId);
+            if (pool.AutoScaleEnabled.HasValue && pool.AutoScaleEnabled.Value)
+            {
+                await this.Client.PoolOperations.DisableAutoScaleAsync(poolId);
+                if (!await this.WaitForResizingOperationToFinish(poolId))
+                {
+                    throw new InvalidOperationException("Autoscale disable operation is already in progress. Please try again after sometime to Resize the pool.");

```

Comment:
This seems a bit arbitrary to me. What is the expected experience in the UI for the 10 seconds we wait? I feel like a better way to do this is just have this as a background operation or fail immediately. Most resize operations take longer than 10 seconds. Delaying this long might be frustrating to the user.